### PR TITLE
Delaney Moran

### PR DESCRIPTION
**Pull request by DelaneyMoran**
round 1

### CODE REVIEW

File: HW-2.Rmd [hunk 1]
```diff
@@ -110,12 +135,208 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
+  geom_histogram() +
+  facet_wrap(~sex, ncol=1) +
+  ggtitle("Age")
+
+#ORIENTATION
+orientations <- select(profiles2, sex, orientation) 
+
+orientations <- mutate(orientations, is_bisexual = ifelse(orientation == "bisexual", 1, 0))
+orientations <- mutate(orientations, is_gay = ifelse(orientation == "gay", 1, 0))
+orientations <- mutate(orientations, is_straight = ifelse(orientation == "straight", 1, 0))
```
Comment (rudeboybert) [inline HW-2.Rmd]: Nitpick: you should thread multiple mutates into a single mutate statement:
```r
orientations <- select(profiles2, sex, orientation) %>% 
  mutate(
    is_bisexual = ifelse(orientation == "bisexual", 1, 0),
    is_gay = ifelse(orientation == "gay", 1, 0),
    is_straight = ifelse(orientation == "straight", 1, 0)
  )
```

File: HW-2.Rmd [hunk 2]
```diff
@@ -110,12 +135,208 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
+            straight = mean(is_straight)) %>% 
+  gather(key = "orientation", value = prop, -sex)
+
+ggplot(prop_orientations, aes(x = sex, y = prop, fill = orientation)) +
+  geom_bar(stat = "identity", position = "stack") +
+  ylab("proportion") +
+  ggtitle("Sexual Orientation by Gender")
+
+#INCOME BY GENDER
+income <- select(profiles2, sex, income) %>%
```
Comment (rudeboybert) [inline HW-2.Rmd]: Good job catching the incomes coded as -1. But recall our discussions yesterday on missing data!

File: HW-2.Rmd [hunk 3]
```diff
@@ -110,12 +135,208 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
 
 ```{r, echo=FALSE, fig.width=12, fig.height=6}
-# Feel free to make multiple code blocks, but set echo, fig.width, fig.height as 
-# above
+#create variable is_female
+profiles <- mutate(profiles, is_female = ifelse(sex == "f", 1, 0))
+
+# Split off the essays into a separate data.frame
+essays <- select(profiles, is_female, contains("essay"))
+profiles2 <- select(profiles, -contains("essay"))
```
Comment (rudeboybert) [inline HW-2.Rmd]: Just note as I said in class a few times, if computation starts slowing down, then work with a random sample of the observations using `sample_n` or `sample_frac` commands during the development phase, and only use the full data at the end.

File: HW-2.Rmd [hunk 4]
```diff
@@ -110,12 +135,208 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
+            skinny = mean(is_skinny),
+            athletic = mean(is_athletic),
+            thin = mean(is_thin),
+            average = mean(is_average),
+            fit = mean(is_fit),
+            extra = mean(is_extra),
+            full = mean(is_full),
+            jacked = mean(is_jacked),
+            declined = mean(is_nope)) %>% 
+  gather(key = "body", value = prop, -sex)
```
Comment (rudeboybert) [inline HW-2.Rmd]: Great use of `tidyr`! yes, to compute proportions like you did, you often need to use `gather` or `spread`. I consider `dplyr` a screwdriver, `ggplot2` a hammer, and `dplyr` is a socket wrench. i.e. the latter may not be used that often, but when you need it, you really need it.

File: HW-2.Rmd [hunk 5]
```diff
@@ -110,12 +135,208 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
+            thin = mean(is_thin),
+            average = mean(is_average),
+            fit = mean(is_fit),
+            extra = mean(is_extra),
+            full = mean(is_full),
+            jacked = mean(is_jacked),
+            declined = mean(is_nope)) %>% 
+  gather(key = "body", value = prop, -sex) 
+
+ggplot(bodytype, aes(x = body, y = prop, fill = sex)) +
```
Comment (rudeboybert) [inline HW-2.Rmd]: Very nice. When choosing predictors, a rough principle I follow is "you want to use predictors that have the most discrimination for the most people".

File: HW-2.Rmd [hunk 6]
```diff
@@ -110,12 +135,208 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
+
+prop_drinking <- drinking %>% 
+  group_by(sex) %>% 
+  summarise(socially = mean(drinks_socially),
+            often = mean(drinks_often),
+            rarely = mean(drinks_rarely),
+            very_often = mean(drinks_voften),
+            not_at_all = mean(drinks_naa)) %>% 
+  gather(key = "drinking", value= prop, -sex)
+
```
Comment (rudeboybert) [inline HW-2.Rmd]: Definitely want to present the ordinal categorical variable in a better order.  This is one thing, as I said yesterday, is a huge drawback of dplyr and ggplot.

This is a hack, but the best I can do it. Run each line here one-by-one:
```r
drinking_levels <- levels(prop_drinking$drinking) %>% dput()
drinking_levels
drinking_levels <- drinking_levels[c(5,3,1,2,4)]
prop_drinking <- prop_drinking %>% 
  mutate(
    drinking = factor(drinking, levels=drinking_levels)
  )
```

File: HW-2.Rmd [hunk 7]
```diff
@@ -110,12 +135,208 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
+#essay contains "child" or "kid"
+essays$has_child <- profile.has.query(data.frame = essays, query = "child")
+essays$has_kid <- profile.has.query(data.frame = essays, query = "kid")
+
+essays <- mutate(essays, kidwords = has_child + has_kid)
+
+kidwords <- group_by(essays, is_female) %>% 
+  summarise(prop_kidwords = mean(kidwords))
+knitr::kable(kidwords, col.names = c("Is Female", "Proportion of Respondents Whose Essay Contains 'Child' or 'Kid'"))
```
Comment (rudeboybert) [inline HW-2.Rmd]: * Be sure to round the digits to something smaller; we don't need the pseudoprecision.
* Also, maybe set the first column to "Sex" and then have the values be "male" and "female"

File: HW-2.Rmd [hunk 8]
```diff
@@ -41,45 +37,76 @@ pollution <- read.dta(url) %>%
   tbl_df()
 ```
 
-### a)
+### a)Create a scatterplot of mortality rate vs. level of nitric oxides. Do you think linear regression will fit these data well? Fit the regression and evaluate a residual plot from the regression.
```
Comment (rudeboybert) [inline HW-2.Rmd]: Please move the question to its own line. The long sequence of bolded text feels like your HW is screaming at me! i.e.:
```r
### a)
Create a scatterplot of mortality rate vs. level of nitric oxides. Do you think linear regression will fit these data well? Fit the regression and evaluate a residual plot from the regression.=
```

**Comment by rudeboybert** [comment]
**OVERALL COMMENTS**: You're on the right track, i.e. you're snooping for variables that could be potentially informative in predicting gender.

Q1 is more of a straightforward textbook question, so I only glanced at this. But things look good.

This is the feedback stage, but some of my comments related to the polishing stage so you should only tackle them once your actual analysis won't change much.
* The comment on combining mutates
* reordering bars

**Comment by DelaneyMoran** [comment]
second upload

**Comment by rudeboybert** [comment]
I am confirming that I received your HW and the .Rmd file knitted correctly.

File: HW-2.Rmd [hunk 9]
```diff
@@ -42,49 +39,113 @@ pollution <- read.dta(url) %>%
...
+b <- coefficients(model1)
+             
+plot1 + 
+  geom_smooth(method = "lm", se = FALSE) + 
+  ggtitle("Mortality Rate by NOx Level") +
+  xlab("NOx level") +
+  ylab("Mortality")
+
+```
+The regression is heavily skewed by the vertical spread of y values at low levels of x, and the few data points at large values of x.
```
Comment (rudeboybert) [inline HW-2.Rmd]: residual plot?

File: HW-2.Rmd [hunk 10]
```diff
@@ -42,49 +39,113 @@ pollution <- read.dta(url) %>%
...
+pollution2 <- mutate(pollution, log_nox = log(nox))
+plot2 <- ggplot(pollution2, aes(x = log_nox, y = mort)) + geom_point()
+
+model2 <- lm(mort ~ log_nox, data = pollution2)
 
+plot2 + 
+  geom_smooth(method = "lm", se = FALSE) +
+  ggtitle("Mortality Rate by NOx Level") +
+  xlab("Log NOx level") +
+  ylab("Mortality Rate")
```
Comment (rudeboybert) [inline HW-2.Rmd]: residual plot? evaluation of it?

File: HW-2.Rmd [hunk 11]
```diff
@@ -42,49 +39,113 @@ pollution <- read.dta(url) %>%
...
 ```
 
+### c) 
+####Interpret the slope coefficient from the model you chose in b.
+
+```{r, echo=FALSE, fig.width=12, fig.height=6}
+coefficients(model2)
+```
```
Comment (rudeboybert) [inline HW-2.Rmd]: I would also add `confint(model2)` to present the confidence intervals. i.e. do they contain 0? are they significant?

File: HW-2.Rmd [hunk 12]
```diff
@@ -42,49 +39,113 @@ pollution <- read.dta(url) %>%
...
+plot3 <- ggplot(pollution2, aes(x = log_so2, y = mort, color = log_nox, size = log_hc)) + 
+  geom_point() +
+  ggtitle("Mortality by Particulate Matter") +
+  xlab("Log SO2 levels") +
+  ylab("Mortality") 
+
+plot3 + geom_smooth(method = "lm", se = FALSE) 
 
 ```
+Each increase of 1 in SO2 level corresponds to an increase of 11.76 in mortality. Similarly, every increase in NOx is associated with an increase of 58.34 in mortality. Surprisingly, hydrocarbons are negatively associated with mortality, such that an increase of 1 in hc is associated with a decrease in mortality of 57.30.
```
Comment (rudeboybert) [inline HW-2.Rmd]: nope. you logged the predictor variables, so the interpretation changes. the interpretation you wrote would be valid had you NOT logged the predictor variables

File: HW-2.Rmd [hunk 13]
```diff
@@ -42,49 +39,113 @@ pollution <- read.dta(url) %>%
...
-### c)
+### d) 
+####Now fit a model predicting mortality rate using levels of nitric oxides, sulfur dioxide, and hydrocarbons as inputs. Use appropriate transformations when helpful. Plot the fitted regression model and interpret the coeffiecients. 
 
 ```{r, echo=FALSE, fig.width=12, fig.height=6}
+pollution2 <- mutate(pollution2, log_so2 = log(so2),
+                     log_hc = log(hc))
+
+model3 <- lm(mort ~ log_nox + log_so2 + log_hc, data = pollution2)
+model3
```
Comment (rudeboybert) [inline HW-2.Rmd]: I will show a better way to present regression tables during discussion on Wednesday.

File: HW-2.Rmd [hunk 14]
```diff
@@ -42,49 +39,113 @@ pollution <- read.dta(url) %>%
...
-### d)
 
 ```{r, echo=FALSE, fig.width=12, fig.height=6}
 
-```
+pollution<- mutate(pollution, log_so2 = log(so2),
+                     log_hc = log(hc),
+                     log_nox = log(nox))
 
+pollutiontrain <- sample_frac(pollution, 0.7)
```
Comment (rudeboybert) [inline HW-2.Rmd]: I'm assuming you were trying to randomly sample the training set, and then ensure the test set was what's left over. You are correct in your thinking, but `as.numeric(rownames(pollutiontrain))` only returns the numbers 1:42. Rather you should've

* created a new variable in `pollution` called `rownumber`
* done the sampling to get `pollutiontrain`
* then used `rownumber` to see what rows to assign to `pollutiontest`

File: HW-2.Rmd [hunk 15]
```diff
@@ -42,49 +39,113 @@ pollution <- read.dta(url) %>%
...
+modelcheck <- lm(mort ~ predictions, data = pollutiontest)
 
+summary(modelcheck)
+modelcheck
+confint(modelcheck)
 
-```{r, echo=FALSE, fig.width=12, fig.height=6}
+polplot <- ggplot(pollutiontest, aes(x = predictions, y = mort)) + 
+  geom_point() +
+  stat_smooth(method = "lm") +
```
Comment (rudeboybert) [inline HW-2.Rmd]: * you shouldn't run a regression, but plot the line y= x via `geom_abline(intercept = 0, slope = 1)`
* I wouldn't use error bars here, as they might be misinterpreted.

File: HW-2.Rmd [hunk 16]
```diff
@@ -42,49 +39,113 @@ pollution <- read.dta(url) %>%
...
 
+pollutiontest <- mutate(pollutiontest, predictions = predict(model4, pollutiontest)) 
 
-### f) What do you think are the reasons for using cross-validation?
 
+modelcheck <- lm(mort ~ predictions, data = pollutiontest)
 
+summary(modelcheck)
+modelcheck
+confint(modelcheck)
```
Comment (rudeboybert) [inline HW-2.Rmd]: I think the plot below would suffice.

File: HW-2.Rmd [hunk 17]
```diff
@@ -42,49 +39,113 @@ pollution <- read.dta(url) %>%
...
+  ggtitle("Prediction Analysis")
+polplot
 
 ```
+If the model worked perfectly, we would expect to see the line y = x when plotting predicted vs observed rates. Depending on the sample fraction used, a better-fitting model can be reliably produced. I was finding very variable slopes when I used only half the dataset to train the model, so switched to 0.7. Still the model changes with each run because it is based off of a random sample of the data. 
 
+### f) 
+####What do you think are the reasons for using cross-validation?
 
+The reason to cross-validate is to make sure the model isn't too specific. With a small data set, it is possible to create a very good model that predicts every data point very closely. However, it is hard to know if such a perfect model is generalizable. One way to test the model is to use only a portion of the data to make the model and then test it on the other portion.
```
Comment (rudeboybert) [inline HW-2.Rmd]: I would add to the first sentence: The reason to cross-validate is to make sure the model isn't too specific TO THE DATASET WE USED TO FIT THE MODEL.

This is called out-of-sample prediction.

File: HW-2.Rmd [hunk 18]
```diff
@@ -110,12 +171,221 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
+            quant2 = quantile(income, (3/4)))
+
+ggplot(income, aes(x = sex, y = income)) + 
+  geom_boxplot(stat = "boxplot") +
+  scale_y_continuous(limits = c(0, 250000)) +
+  coord_flip() +
+  ggtitle("Income by Gender")
+
+```
+Note that height and income differ by gender, while age and sexual orientation are pretty similar.
```
Comment (rudeboybert) [inline HW-2.Rmd]: Really about orientation? Remember, this is SF.  Look at the proportion of gay males vs gay females. They only look similar at first glance b/c the straight category is so big.

Your treatment of income is a little dangerous. There were A LOT of missing values. If someone came up to you and asked you for your income, would you volunteer this info liberally? You should at least report what proportion of males and females didn't report their income. That's what missing data means in this context.

File: HW-2.Rmd [hunk 19]
```diff
@@ -110,12 +171,221 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
+
+
+###More Personal:
+```{r, echo=FALSE, fig.width=12, fig.height=6}
+#BODY TYPE BY GENDER
+bodytype <- profiles2 %>% 
+  select(body_type, sex)
+
+bodytype<- mutate(bodytype, is_curvy = ifelse(body_type == "curvy", 1, 0),
```
Comment (rudeboybert) [inline HW-2.Rmd]: you forgot `used_up` here

File: HW-2.Rmd (hunk 5 above)
Comment (rudeboybert) [inline HW-2.Rmd]: Again, good job `group_by` sex and then show proportions in each sex. A lot of people only showed raw counts. but becuase there are wayyyyy more males in this data set its not fair to compare counts straight up

File: HW-2.Rmd (hunk 6 above)
Comment (rudeboybert) [inline HW-2.Rmd]: You didn't reorder the categories here to reflect that not_all_all < rarely < socially < often < very_often

File: HW-2.Rmd [hunk 20]
```diff
@@ -110,12 +171,221 @@ profiles <- read.csv("profiles.csv", header=TRUE) %>% tbl_df()
...
+
+#essay contains "child" or "kid"
+essays$has_child <- profile.has.query(data.frame = essays, query = "child")
+essays$has_kid <- profile.has.query(data.frame = essays, query = "kid")
+
+essays <- mutate(essays, kidwords = has_child + has_kid)
+
+kidwords <- group_by(essays, sex) %>% 
+  summarise(prop_kidwords = mean(kidwords))
+knitr::kable(kidwords, digits = 3, col.names = c("Sex", "Proportion of Respondents Whose Essay Contains 'Child' or 'Kid'"))
```
Comment (rudeboybert) [inline HW-2.Rmd]: Nice clean tables. Not a big deal, but and alternate way to display this is in a single table, where the column names would be "Read" "Cook" "Cars, truck or motorcycle" and "Child Kid".

**Comment by rudeboybert** [comment]
**overall comments**: you forgot your residuals plots. also it really helps aid reading graphs if ordinal categorical variables are presented in their hierarchical order. Other than that, you're set of HW-3.

Comments best viewed [side-by-side](https://github.com/Middlebury-Data-Science/HW-2/pull/2/files)

File: HW-2.Rmd [hunk 21]
```diff
@@ -15,17 +17,12 @@ suppressPackageStartupMessages(library(foreign))
 
 Please indicate
 
-* Who you collaborated with:
-* Roughly how much time you spent on this HW:
+* Who you collaborated with: none
```
Comment (rudeboybert) [inline HW-2.Rmd]: if you work with people, please acknowledge this.
